content/chat/participant-activity.textile

```diff
@@ -1,7 +1,220 @@
+
+```[rest_javascript]
+var ablyRest = new Ably.Rest({ key: '{{API_KEY}}' })
+var content = { "channel": "channel1,channel2" }
+ablyRest.request('GET', '/presence', content, null, {}, function(err, response) {
+  If (err) {
+    alert('An error occurred; err = ' + err.toString());
+  } else {
+    alert('Success! status code was ' + response.statusCode);
+  }
+});
+```
+
+h4(#batch-responses). Batch presence responses
+
+On completion of the batch request, one of three possible responses are returned:
+
+- Success := If all of the individual requests were successful then an array containing the response of each query is returned in request order.
+- Failure := If the batch request itself failed before the individual requests were made, then an error response is returned with a status code and error response body. Examples of why the batch request can fail include an authorization failure or an invalid request.
+- Partial success := If one or more of the individual requests failed the response body contains an error object with the error code @40020@ and a status code of @400@. The error body contains a @batchResponse@ array of each individual response in request order. The @batchResponse@ can be inspected if there is a need to know the details of each outcome. If you only need to know whether or not the batch request was completely successful then the status code is sufficient.
+
+The following is an example of a successful batch presence response. The response body contains details of each client present on the channel and the channel they are present on. The status code for a successful response is always @200@:
+
+```[json]
+[
+  {
+    "channel":"channel0",
+    "presence":[
+      {"clientId": "user1", "action": "1"},
+      {"clientId": "user2", "action": "1"}
+    ]
+  },
+  {
+    "channel":"channel1",
+    "presence":[]
+  },
+  {
+    "channel":"channel2",
+    "presence":[
+      {"clientId": "user2", "action": "1"},
+      {"clientId": "user3", "action": "1"}
+    ]
+  }
+]
+```
+
+For more information making batch presence requests please refer to the "Presence documentation":/docs/presence-occupancy/presence#batch
+
+
+h3(#channel-separation). Channel separation
+
+Chat messages and chat participant status updates in one pub/sub channel would contain a large number of messages published and received. To avoid a potentially high volume of unnecessary messages, as well as the suggestion of only having your backend server being the exclusive subscriber to chat participants' online status updates, it is also advised to have separate pub/sub channels for separate purposes. One for a specific chat (e.g. general), while having a separate channel for the chat participants status (e.g. status-general) updates for that chat. For example, when issuing api keys to clients, assigning the clients the following "capabilities":https://ably.com/docs/auth/capabilities :
```

Comment:
This is a good opportunity to expand on typical use case scenario e.g. live streaming and where separate channels makes sense

Think this is the first time we have state "pub/sub channel" - just make "channel"?

Reply:
This paragraph has been made clearer, I've removed the "pub/sub" part, and also included references to the use case scenario of large-scale live streaming